components/pluralrules/src/lib.rs

```diff
@@ -0,0 +1,381 @@
+//!
+//! * `1 item`
+//! * `0 items`, `2 items`, `5 items`, `0.5 items` etc.
+//!
+//! The former variant is used when the placeholder variable has value `1`,
+//! while the latter is used for all other values of the variable.
+//!
+//! Unicode defines [`Language Plural Rules`] as a mechanism to codify those
+//! variants and provides data and algorithms to calculate
+//! appropriate [`Plural Category`].
+//!
+//! # Examples
+//!
+//! ```
+//! use icu_locale::LanguageIdentifier;
+//! use icu_pluralrules::{PluralRules, PluralRuleType, PluralCategory};
+//! use icu_pluralrules::data::provider::DummyDataProvider;
+//!
+//! let lang: LanguageIdentifier = "en".parse()
+//!     .expect("Failed to parse a language identifier.");
+//!
+//! let dp = DummyDataProvider::default();
+//!
+//! let pr = PluralRules::try_new(lang, PluralRuleType::Cardinal, &dp)
+//!     .expect("Failed to construct a PluralRules struct.");
+//!
+//! assert_eq!(pr.select(1_usize), PluralCategory::One);
+//! assert_eq!(pr.select(5_usize), PluralCategory::Other);
+//! ```
+//!
+//! ## Plural Rules
+//!
+//! The crate provides the main struct [`PluralRules`] which handles selection
+//! of the correct [`Plural Category`] for a given language and [`Plural Type`].
+//!
+//! ## Plural Category
+//!
+//! Every number in every language belongs to a certain [`Plural Category`].
+//! For example, Polish language uses four:
+//!
+//! * [`One`](./enum.PluralCategory.html#variant.One): `1 miesiąc`
+//! * [`Few`](./enum.PluralCategory.html#variant.Few): `2 miesiące`
+//! * [`Many`](./enum.PluralCategory.html#variant.Many): `5 miesięcy`
+//! * [`Other`](./enum.PluralCategory.html#variant.Other): `1.5 miesiąca`
+//!
+//! ## Plural Rule Type
+//!
+//! Plural rules depend on the use case. This crate supports two types of plural rules:
+//!
+//! * [`Cardinal`](./enum.PluralRuleType.html#variant.Cardinal): `3 doors`, `1 month`, `10 dollars`
+//! * [`Ordinal`](./enum.PluralRuleType.html#variant.Ordinal): `1st place`, `10th day`, `11th floor`
+//!
+//! ## Data Provider
+//!
+//! In order to function, the API requires data from [`CLDR`].
+//!
+//! [`ICU4X`] is going to use a special API for handling data management called `DataProvider`.
+//! Until that happens, this crate will provide a simple `JSON` and `bincode` providers behind a
+//! flag.
+//! For tests and documentation examples, there is also a `DummyDataProvider`.
+//!
+//! All of the content of the [`data`] module is heavily experimental and subject to change.
+//!
+//! [`ICU4X`]: https://github.com/unicode-org/icu4x
+//! [`PluralRules`]: ./struct.PluralRules.html
+//! [`PluralRules`]: ./struct.PluralRules.html
+//! [`Plural Type`]: ./enum.PluralRuleType.html
+//! [`Plural Category`]: ./enum.PluralCategory.html
+//! [`Language Plural Rules`]: https://unicode.org/reports/tr35/tr35-numbers.html#Language_Plural_Rules
+//! [`CLDR`]: http://cldr.unicode.org/
+//! [`data`]: ./data/index.html
+pub mod data;
+mod operands;
+pub mod rules;
```

Comment:
Why pub?

Reply:
Same as `ast` - I need it for tests/benchmarks.